### PR TITLE
Tap 5239 add catch data feature

### PR DESCRIPTION
**Pull request by openlg**

### CODE REVIEW

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **58.4**
- Status: **ERROR**, MetricKey: **new_duplicated_lines_density**, ActualValue: **3.15789**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=TAP-5239-feature-debug-data&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **76.2**
- Status: **ERROR**, MetricKey: **new_duplicated_lines_density**, ActualValue: **3.15789**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=TAP-5239-feature-debug-data&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **60.9**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=TAP-5239-feature-debug-data&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **71.7**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=TAP-5239-feature-debug-data&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **70.8**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=TAP-5239-feature-debug-data&id=tapdata

**Comment by xbsura** [comment]
SonarQube Quality Gate Status: **ERROR**

- Status: **ERROR**, MetricKey: **new_coverage**, ActualValue: **67.4**


See Sonar Scan Result at: http://58.251.34.123:29000/dashboard?branch=TAP-5239-feature-debug-data&id=tapdata

File: iengine/iengine-app/src/main/java/io/tapdata/flow/engine/V2/node/hazelcast/HazelcastBaseNode.java [hunk 1]
```diff
@@ -475,6 +474,48 @@ protected boolean offer(TapdataEvent dataEvent) {
...
+					processorBaseContext.getTaskDto().getId().toHexString(),
+					Optional.ofNullable(processorBaseContext.getTaskDto().getLogSetting())
+							.map(m -> m.get("query"))
+							.filter(Objects::nonNull)
+							.map(Object::toString).orElse(null));
+		}
+
+		if (!dataEvent.isCatchMe())
+			return;
+		List<Map<String, Object>> data = new ArrayList<>(2);
```
Comment (ply0011) [inline iengine/iengine-app/src/main/java/io/tapdata/flow/engine/V2/node/hazelcast/HazelcastBaseNode.java:489]: 使用TapEventUtil获取after或者before

File: iengine/modules/observable-module/src/main/java/io/tapdata/observable/logging/ObsLoggerFactory.java [hunk 2]
```diff
@@ -53,7 +56,7 @@ private ObsLoggerFactory() {
 		this.settingService = Optional.ofNullable(BeanUtil.getBean(SettingService.class)).orElse((SettingService) getBeanAsync(SettingService.class));
 		this.clientMongoOperator = Optional.ofNullable(BeanUtil.getBean(ClientMongoOperator.class)).orElse((ClientMongoOperator) getBeanAsync(ClientMongoOperator.class));
 		this.scheduleExecutorService = new ScheduledThreadPoolExecutor(1);
-		scheduleExecutorService.scheduleAtFixedRate(this::renewTaskLogSetting, 0L, PERIOD_SECOND, TimeUnit.SECONDS);
+		//scheduleExecutorService.scheduleAtFixedRate(this::renewTaskLogSetting, 0L, PERIOD_SECOND, TimeUnit.SECONDS);
```
Comment (ply0011) [inline iengine/modules/observable-module/src/main/java/io/tapdata/observable/logging/ObsLoggerFactory.java:59]: 不需要的代码删除

File: iengine/modules/observable-module/src/main/java/io/tapdata/observable/logging/ObsLoggerFactory.java [hunk 3]
```diff
@@ -86,6 +89,11 @@ Object getBeanAsync(Class<?> clz){
 	public Map<String, TaskLogger> getTaskLoggersMap() {
 		return taskLoggersMap;
 	}
+
+	/**
+	 * log.debug used to catch data and catch data switch use @io.tapdata.services.CatchDataService to control
+	 */
+	@Deprecated
```
Comment (ply0011) [inline iengine/modules/observable-module/src/main/java/io/tapdata/observable/logging/ObsLoggerFactory.java:96]: 可以删掉了

**Review by ply0011 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**